schema/workflow.yaml

```diff
@@ -1196,14 +1196,41 @@ $defs:
               $ref: '#/$defs/eventFilter'
         required: [ all ]
       - title: AnyEventConsumptionStrategy
-        properties:
-          any:
-            type: array
-            title: AnyEventConsumptionStrategyConfiguration
-            description: A list containing any of the events to consume.
-            items:
-              $ref: '#/$defs/eventFilter'
-        required: [ any ]
+        oneOf:
+          - properties:
+              any:
+                type: array
+                title: AnyEventConsumptionStrategyConfiguration
+                description: A list containing any of the events to consume.
+                items:
+                  $ref: '#/$defs/eventFilter'
+                minItems: 1
+              until:
+                oneOf:
+                  - type: string
+                    title: AnyEventUntilCondition
+                    description: A runtime expression condition evaluated after consuming an event and which determines whether or not to continue listening.
+                  - $ref: '#/$defs/eventConsumptionStrategy'
```

Comment:
If I understood correctly, we would have to create a new `EventConsumption` type here to avoid the infinity mirror effect. 😆

Reply:
Precisely!!!